include/natalie/value.hpp

```diff
@@ -143,6 +136,18 @@ class Value {
             hydrate();
     }
 
+    bool fast_equal(Value other) const {
+        if (is_fast_integer() && other.is_fast_integer())
+            return get_fast_integer() == other.get_fast_integer();
+        else if (is_fast_integer() && other.is_fast_float())
+            return get_fast_integer() == other.get_fast_float();
```

Comment:
I just realized this is wrong. The `operator==` check is to check if the objects are the _same pointer_ -- not logically equivalent. A float and an int would never _be the same pointer_.